File: Laba4/Java/Philosopher.java
class Philosopher implements Runnable {
    private final Table table;
    private final int id;
    private final Thread thread;
    private final Strategy strategy;

    public Philosopher(int id, Table table, Strategy strategy) {
        this.id = id;
        this.table = table;
        this.strategy = strategy;

        this.thread = new Thread(this);
        this.thread.start();
    }

    @Override
    public void run() {
        // ANSI Color Codes
        String cyan = "\u001B[36m";
        String yellow = "\u001B[33m";
        String reset = "\u001B[0m";

        for (int i = 0; i < 10; i++) {
            synchronized (Main.lock) {
                System.out.println(cyan + "Philosopher" + id + " is thinking " + (i + 1) + " times" + reset);
            }

            strategy.takeForks(id, table);

            synchronized (Main.lock) {
                System.out.println(yellow + "Philosopher" + id + " is eating " + (i + 1) + " times" + reset);
            }

            strategy.putForks(id, table);
        }
    }
}

File: Laba4/Sharp/Philosopher.cs
namespace Laba4
{
    public class Philosopher
    {
        private readonly Table table;
        private readonly int id;
        private readonly Thread thread;
        private readonly IStrategy strategy;

        public Philosopher(int id, Table table, IStrategy strategy)
        {

            this.id = id;
            this.table = table;
            this.strategy = strategy;

            thread = new Thread(Start);
            thread.Start();
        }

        public void Start()
        {
            for (int i = 0; i < 10; i++)
            {
                lock (Program.lock_)
                {
                    Console.ForegroundColor = ConsoleColor.DarkCyan;
                    System.Console.WriteLine("Philosopher" + id + " is thinking " + (i + 1) + " times");
                    // Thread.Sleep(Random.Shared.Next(10, 100));
                    Console.ResetColor();
                }

                strategy.TakeForks(id, table);

                lock (Program.lock_)
                {
                    Console.ForegroundColor = ConsoleColor.DarkYellow;
                    System.Console.WriteLine("Philosopher" + id + " is eating " + (i + 1) + " times");
                    // Thread.Sleep(Random.Shared.Next(10, 100));
                    Console.ResetColor();
                }

                strategy.PutForks(id, table);
            }
        }

    }
}